### PR TITLE
Convert to use arcpy instead of arcgisscripting.

### PR DESCRIPTION
**Pull request by tomironkko**
Migrated to use use arcpy instead of arcgisscripting. Also, improved comments in code.

### CODE REVIEW

**Review by tomironkko — COMMENTED** [review]
Area Freaquency tool migrated. Also, other related files migated to use arcpy.

File: Toolbox/arcsdm/areafrequency.py [hunk 1]
```diff
@@ -39,369 +38,257 @@
...
+            tblrows.updateRow(tblrow)
             effarea.append(area)
             nSites.append(frequency)
             stats_found += 1
-            # gp.AddMessage("Debug: Stats_found =  %s"%stats_found);
         
-        tblrow = tblrows.Next()
+        tblrow = tblrows.next()
     
-    del tblrow,tblrows
```
Comment (lehtonenp) [inline Toolbox/arcsdm/areafrequency.py:291]: I would replace this part with 
`    totalarea = 0.0
    effarea = []
    nSites = []

    with arcpy.da.UpdateCursor(Output_Table, ["RASTERVALU", "Area_sqkm", "Frequency"]) as cursor:
        for row in cursor:
            tblval = row[0]
            area = row[1]
            totalarea += area
            rasval = flt_ras[tblval]
            if rasval in stats_dict:
                frequency = stats_dict[rasval]
                row[2] = frequency
                cursor.updateRow(row)
                effarea.append(area)
                nSites.append(frequency)
                stats_found += 1`

File: Toolbox/arcsdm/areafrequency.py [hunk 2]
```diff
@@ -39,369 +38,257 @@
...
         tblrow.Cum_Sites = Cum_Sites
-        tblrow.SetValue('I_CumSites', 100.0 - Cum_Sites)
-        tblrows.UpdateRow(tblrow)
-        tblrow = tblrows.Next()
-    # gp.addmessage('done.')
+        tblrow.setValue('I_CumSites', 100.0 - Cum_Sites)
+        tblrows.updateRow(tblrow)
+        tblrow = tblrows.next()
+    
     del tblrow, tblrows
```
Comment (lehtonenp) [inline Toolbox/arcsdm/areafrequency.py:258]: I would replace this with:
```
    with arcpy.da.UpdateCursor(Output_Table, ["RASTERVALU", "Area_sqkm", "CAPP_CumAr", "Eff_CumAre", "Cum_Sites", "I_CumSites"]) as cursor:
        for row in cursor:
            cumArea += 100.0 * row[1] / totalarea
            row[2] = cumArea
            row[3] = next(effCumareaList_rev)
            Cum_Sites = next(cumSitesList_rev)
            row[4] = Cum_Sites
            row[5] = 100.0 - Cum_Sites
            cursor.updateRow(row)
```

File: Toolbox/arcsdm/areafrequency.py [hunk 3]
```diff
@@ -39,369 +38,257 @@
...
-    #     # return gp messages for use with a script tool
-    #     gp.AddError(pymsg)
-
-    #     # print messages for use in Python/PythonWin
-    #     print (pymsg)
-    #     print (msgs)
-    #     raise
+        tblrows1.updateRow(tblrow1)
+    
+    del tblrow1, tblrows1
```
Comment (lehtonenp) [inline Toolbox/arcsdm/areafrequency.py:288]: I would replace this with:
```
    with arcpy.da.SearchCursor(Output_Table, ["Eff_CumAre", "Cum_Sites"]) as cursor:
        for row in cursor:
            Eff_CumAre.append(row[0])
            Cum_Sites.append(row[1])
    
    sumEff_AUC = 0.0
    with arcpy.da.UpdateCursor(Output_Table, ["Eff_CumAre", "Cum_Sites", "Eff_AUC"]) as cursor:
        for i, row in enumerate(cursor):
            if i < len(Eff_CumAre):
                val = 0.5 * (row[0] - Eff_CumAre[i]) * (row[1] + Cum_Sites[i]) / (100.0 * 100.0)
                sumEff_AUC += val
                row[2] = val
                cursor.updateRow(row)
            else:
                val = 0.5 * row[0] * row[1] / (100.0 * 100.0)
                sumEff_AUC += val
                row[2] = val
                cursor.updateRow(row)
```

**Review by lehtonenp — APPROVED** [review]
The code works as intended but here are suggestions to improve the code.
With the changes we don't need to worry about the deletion of variables and the code follows ESRI's suggested syntax.

File: Toolbox/arcsdm/areafrequency.py (hunk 1 above)
Comment (tomironkko) [reply Toolbox/arcsdm/areafrequency.py]: I have edited the code and commited new code.

File: Toolbox/arcsdm/areafrequency.py (hunk 2 above)
Comment (tomironkko) [reply Toolbox/arcsdm/areafrequency.py]: I have edited the code and commited new code.

File: Toolbox/arcsdm/areafrequency.py (hunk 3 above)
Comment (tomironkko) [reply Toolbox/arcsdm/areafrequency.py]: I have edited the code and commited new code.

**Pull request merged**